src/main/java/jenkins/branch/BranchNameContributor.java

```diff
@@ -57,10 +58,14 @@ public void buildEnvironmentFor(Job j, EnvVars envs, TaskListener listener) thro
             BranchProjectFactory projectFactory = ((MultiBranchProject) parent).getProjectFactory();
             if (projectFactory.isProject(j)) {
                 Branch branch = projectFactory.getBranch(j);
+                SCMRevision rev = projectFactory.getRevision(j);
                 SCMHead head = branch.getHead();
                 // Note: not using Branch.name, since in the future that could be something different
                 // than SCMHead.name, which is what we really want here.
                 envs.put("BRANCH_NAME", head.getName());
+                if (rev != null) {
+                    envs.put("CHANGE_REVISION", rev.toString());
```

Comment:
I'm not sure that `rev.toString()` is the right thing here, as it is intended to be 
["human-readable"](https://github.com/jenkinsci/scm-api-plugin/blob/master/src/main/java/jenkins/scm/api/SCMRevision.java#L68).  It may work in some of implementations but it, I'm not sure.

It looks like `SCMRevision` doesn't have a method for what you're trying to do.  There probably needs to be a  `getId()` method added to that in order for this to work reliably. Note, we'll also need to handle [ChangeRequestSCMRevision](https://github.com/jenkinsci/scm-api-plugin/blob/master/src/main/java/jenkins/scm/api/mixin/ChangeRequestSCMRevision.java) differently since it has several possible revisions that users would be interested in.  

It might be better if each SCM source implemented their own environment contributor.